osmnx/io.py

```diff
@@ -250,17 +267,23 @@ def _convert_edge_attr_types(G, node_type):
         input graph
     node_type : type
         convert osmid to this type
+    edge_dtypes : dict of attribute name -> data type
+        identifies additional is a numpy.dtype or Python
+        type to cast one or more additional edge attributes.
+        Defaults to {"length": float, "grade": float,
+        "grade_abs": float, "bearing": float,
+        "speed_kph": float,"travel_time": float} if None
 
     Returns
     -------
     G : networkx.MultiDiGraph
     """
-    # convert numeric, bool, and list edge attributes from string to correct data types
+    if edge_dtypes is None:
+        edge_dtypes = {"length": float, "grade": float, "grade_abs": float,
+                       "bearing": float, "speed_kph": float, "travel_time": float}
+    # convert numeric, bool, and list edge attributes from string
+    # to correct data types
```

Comment:
By adding this `if` condition, you increased the complexity from 15 to 16 ([link](https://www.flake8rules.com/rules/C901.html)). As I don't see an obvious way to reduce the complexity of `_convert_edge_attr_types` I see two ways to bypass this test:

1. Increase the `flake8` complexity threshold (Not sure how to do that)
2. Split the `_convert_edge_attr_types` function.

Reply:
I find it funny an urban planning inspired library has the same "last one in pays" issues faced by real estate developers in redevelopment markets. Software developers were naïve to think they were immune!
I vote to increase the flake8 complexity slightly here. The function was already complex, and I think I already contributed to make it more readable. I am not sure breaking it up further makes sense to a human reading it. Including the doc string I might have to add another 6-10 lines to the code base to accommodate the test. I am also content to just sit on the PR for a while to see if any consensus develops.

Reply:
I'll have some time to take a look in the coming couple of days. I can make some revisions as needed from there. Thanks for the contribution.

Reply:
Sounds good! Thanks Geoff!